Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1342,39 +1339,296 @@ namespace OpenInfraPlatform
+					//	https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/lexical/IfcOffsetCurveByDistances.htm
+					//	ENTITY IfcOffsetCurveByDistances
+					//		SUBTYPE OF(IfcOffsetCurve);
+					//			OffsetValues: LIST[1:? ] OF IfcPointByDistanceExpression;
+					//			Tag: OPTIONAL IfcLabel;
+					//	END_ENTITY;
+					// **************************************************************************************************************************
+
+					EXPRESSReference<typename IfcEntityTypesT::IfcCurve> basisCurve = offsetCurveByDistances->BasisCurve;
+					std::vector<carve::geom::vector<3>> basisCurvePoints;
+
+					convertIfcCurve(basisCurve, basisCurvePoints, segmentStartPoints,
+						trim1Vec, trim2Vec, senseAgreement, trimmingPreference);
+
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcPointByDistanceExpression>> OffsetValues = offsetCurveByDistances->OffsetValues;
+					std::vector < std::vector<carve::geom::vector<3>>> offsetCurves;
+
+					for (auto OffsetValue : OffsetValues) {
+						std::vector<carve::geom::vector<3>> offsetCurvePoints;
+						double distance = 0.;
+						switch (OffsetValue->DistanceAlong.which())
+						{
+						case 0:
+						{
+							// Calculate the length using \c IfcNonNegativeLengthMeasure. 
+							distance = OffsetValue->DistanceAlong.get<0>();// *this->UnitConvert()->getLengthInMeterFactor();
+							break;
+						}
+						case 1:
+						{
+							// Calculate the length using \c IfcParameterValue.
+							distance = OffsetValue->DistanceAlong.get<1>();
+							break;
+						}
+						default:
+							throw oip::InconsistentGeometryException(OffsetValue, "Could not determine DistanceAlong!");
+						}
+						
+						double offsetLateral = OffsetValue->OffsetLateral * this->UnitConvert()->getLengthInMeterFactor();
+						double offsetVertical = OffsetValue->OffsetVertical * this->UnitConvert()->getLengthInMeterFactor();
+						double offsetLongitudinal = OffsetValue->OffsetLongitudinal * this->UnitConvert()->getLengthInMeterFactor();
+
+
+						for (int i = 0; i < basisCurvePoints.size() - 1; i++) {
+							carve::geom::vector<3> startPoint = basisCurvePoints[i];
+							carve::geom::vector<3> endPoint = basisCurvePoints[i + 1];
+							carve::geom::vector<3> diretionOfSegment = endPoint - startPoint;
+							carve::geom::vector<3> offsetPoint;
+
+							carve::geom::vector<2> startPoint2D = carve::geom::VECTOR(startPoint.x, startPoint.y);
+							carve::geom::vector<2> endPoint2D = carve::geom::VECTOR(endPoint.x, endPoint.y);
+
+							carve::geom::vector<2> tangent = endPoint2D - startPoint2D;
+							carve::geom::vector<2> orthogonal = carve::geom::VECTOR(-tangent.y, tangent.x);
+
+							offsetPoint = carve::geom::VECTOR(startPoint.x + diretionOfSegment.x * distance + offsetLateral * orthogonal.x,
+								startPoint.y + diretionOfSegment.y * distance + offsetLateral * orthogonal.y,
+								startPoint.z + diretionOfSegment.z * distance + offsetVertical);

```

Comment:
This needs to be adapted 
Use DistanceAlong the way we discussed it

Reply:
Possible solution: [e4565ca](https://github.com/tumcms/Open-Infra-Platform/pull/558/commits/e4565ca74f0e961fa1a01a283695f6f60e536c07)